pkg/acsengine/defaults.go

```diff
@@ -584,8 +591,25 @@ func setMasterProfileDefaults(a *api.Properties, isUpgrade bool) {
 	}
 }
 
-// setVMSSDefaults
-func setVMSSDefaults(a *api.Properties) {
+// setVMSSDefaultsForMasters
+func setVMSSDefaultsForMasters(a *api.Properties) {
+	if a.MasterProfile.Count > 100 {
+		a.MasterProfile.SinglePlacementGroup = helpers.PointerToBool(false)
+	}
+	if a.MasterProfile.SinglePlacementGroup == nil {
+		a.MasterProfile.SinglePlacementGroup = helpers.PointerToBool(api.DefaultSinglePlacementGroup)
+	}
+	if a.MasterProfile.SinglePlacementGroup == helpers.PointerToBool(false) {
+		a.MasterProfile.StorageProfile = api.ManagedDisks
```

Comment:
Is this going to silently override the user's choice if they select storage account? Should we validate + error if the user selects storageAccount + singlePlacement `false`

Reply:
I personally think this is too much detail for end users. so we should set for them but I am ok with either. What would you recommend?

Reply:
The foundation of acs-engine is that it is highly customizable so I think we should assume that anything that is exposed to the user can and will be customized by users. If we want to set a value for the users we shouldn't expose it to them in the apimodel. I think a simple validation check should be enough in this case. That's my recommendation but I'll let you make the call.

Reply:
made it a validation instead of default